classes/Langchecker/DotLangParser.php

```diff
@@ -127,24 +145,31 @@ public static function parseFile($path, $reference_locale = false)
                         $dotlang_data['duplicates'][] = $reference;
                     }
 
-                    if (Utils::startsWith($translation, [';', '#'])) {
-                        /* Empty translation: what I'm reading as translation is either the next reference string
-                         * or the next meta tag (comment, tag binding). I consider this string untranslated.
-                         */
-                        $dotlang_data['strings'][$reference] = $reference;
+                    // If we don't extract metadata, we shouldn't test for #
+                    $markers = self::$extract_metadata
+                        ? [';', '#']
+                        : [';'];
+
+                    // We test the most common scenario first: reference string followed by a translation
+                    if (! Utils::startsWith($translation, $markers)) {
+                        // Store the translation
+                        $dotlang_data['strings'][$reference] = $translation;
```

Comment:
Should we set `$translation` in this conditional, considering that it's now used only here?